tests/daemon/unit/models/good_ws_multiple_files/.jinad

```diff
@@ -1,3 +1,4 @@
 build = devel
 python = 3.7
 run = "echo Hello"
+ports = 12345,123456
```

Comment:
could you add tests where
- `ports` is not passed
- `ports` has a value same as Flow's `port_expose` (there would be a port conflict then, we should probably have a different error for this).

Reply:
I think empty is tested already as part of this parameterize:
`('good_ws_emptyfile', ('devel', '3.8', '', [])),`

The port conflict is a bit more complicated as this happens at two different times? The custom container gets started on workspace creation. The flow will be started later by a separat POST to /flows endpoint? And the same goes probably for all exposed ports between all entities in a workspace?

Reply:
yah. This is a bigger problem that needs to be handled in several places.